### PR TITLE
[stable/keycloak] Update pullSecrets variable usage

### PR DESCRIPTION
**Pull request by alivespirit**
#### What this PR does / why we need it:
Updating pullSecrets variable to match the example in values.yaml. Current version has this example:
```
    pullSecrets: []
    #  - myRegistrKeySecretName
```
but the construction in [templates/statefulset.yaml](templates/statefulset.yaml) expects it to be:
```
    pullSecrets:
      - name: myRegistrKeySecretName
```
So, if pull secrets are set like in values.yaml, deploy fails with error like `Spec: v1.PodSpec: ImagePullSecrets: []v1.LocalObjectReference: readObjectStart: expect { or n, parsing 2503 ...ecrets"`
My changes update template to use pull secret value in a similar way as in stable/postgresql chart.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] pullSecret variable usage updated
- [x] Chart Version bumped
- [x] DCO signed

### CODE REVIEW

**Comment by unguiculus** [comment]
I'd prefer to keep it a it is. The sample in the comment in `values.yaml` should be fixed.

**Comment by alivespirit** [comment]
@unguiculus But this way we are going to have inconsistency in values if we want to set pull secrets for both keycloak and postgres. E.g.:
```
keycloak:
  image:
    pullSecrets:
      - name: myRegistrKeySecretName
postgresql:
  image:
    pullSecrets:
      - myRegistrKeySecretName
```
I think we should stick to one standard.

File: stable/keycloak/values.yaml [hunk 1]
```diff
@@ -18,8 +18,8 @@ keycloak:
     ## Secrets must be manually created in the namespace.
     ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
     ##
-    pullSecrets: []
-    #  - myRegistrKeySecretName
+    # pullSecrets:
```
Comment (unguiculus) [inline stable/keycloak/values.yaml]: Keep it as an empty list

File: stable/keycloak/templates/statefulset.yaml [hunk 2]
```diff
@@ -32,9 +32,11 @@ spec:
     spec:
       securityContext:
 {{ toYaml .Values.keycloak.securityContext | indent 8 }}
-    {{- with .Values.keycloak.image.pullSecrets }}
+    {{- if .Values.keycloak.image.pullSecrets }}
```
Comment (unguiculus) [inline stable/keycloak/templates/statefulset.yaml]: Keep `with` instead of `if` as it was.

**Review by unguiculus — CHANGES_REQUESTED** [review]
OK, accepted. Then let's go ahead and change this. Please consider my comments.

File: stable/keycloak/templates/statefulset.yaml (hunk 2 above)
Comment (alivespirit) [reply stable/keycloak/templates/statefulset.yaml]: Tested:
1)
```
    {{- with .Values.keycloak.image.pullSecrets }}
      imagePullSecrets:
      - name: {{ . }}
    {{- end }}
```
`ImagePullSecrets: []v1.LocalObjectReference: v1.LocalObjectReference: Name: ReadString: expects " or n`
2)
```
    {{- with .Values.keycloak.image.pullSecrets }}
      imagePullSecrets:
      {{- range .Values.keycloak.image.pullSecrets }}
      - name: {{ . }}
      {{- end }}
    {{- end }}
```
`can't evaluate field Values in type []interface {}`  

Works if set to this:
```
    {{- with .Values.keycloak.image.pullSecrets }}
      imagePullSecrets:
    {{- range $.Values.keycloak.image.pullSecrets }}
      - name: {{ . }}
    {{- end }}
    {{- end }}
```
But I'm actually missing the point of keeping `with` in this case, it makes things a bit more complicated.

Comment (unguiculus) [reply stable/keycloak/templates/statefulset.yaml]: From the top of my head, this should work:

```yaml
{{- with .Values.keycloak.image.pullSecrets }}
  imagePullSecrets:
  {{- range . }}
    - name: {{ . }}
  {{- end }}
{{- end }}
```

**Comment by Dean-Coakley** [comment]
@alivespirit If you set 
```
pullSecrets: []
      # - name: myRegistrKeySecretName
```

Then I think you could do:
```
{{- with .Values.keycloak.image.pullSecrets }}
      imagePullSecrets:
{{ toYaml . | indent 8 }}
{{- end }}
```

What do you think?

File: stable/keycloak/templates/statefulset.yaml (hunk 2 above)
Comment (alivespirit) [reply stable/keycloak/templates/statefulset.yaml]: You're right, that works, pushing an update.

**Comment by alivespirit** [comment]
@Dean-Coakley yes, I can do that, but as I mentioned above - than we'll have inconsistency if we would want to use pull secrets for both keycloak and dependent postgresql.

File: stable/keycloak/values.yaml (hunk 1 above)
Comment (alivespirit) [reply stable/keycloak/values.yaml]: Pushed an update.

**Review by Dean-Coakley — APPROVED** [review]
I agree. Looks good now.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**